src/System.Private.Xml/src/System/Xml/Core/XmlTextWriter.cs

```diff
@@ -338,15 +343,45 @@ public int Indentation
             {
                 if (value < 0)
                     throw new ArgumentException(SR.Xml_InvalidIndentation);
-                _indentation = value;
+                if(_indentation != value)
+                {
+                    _indentation = value;
+                    int arraySize = _indentation * IndentArrayLength;
+                    if (arraySize > _indentChars.Length)
+                    {
+                        char indentChar = _indentChars[0];
+                        _indentChars = new char[arraySize];
```

Comment:
Agreed

---
In reply to: [122556017](https://github.com/dotnet/corefx/pull/21155#discussion_r122556017) [](ancestors = 122556017)